web/modules/custom/eulogin/tests/src/ExistingSite/JoinupEuLoginTest.php

```diff
@@ -128,16 +130,31 @@ public function testLimitedAccess(): void {
     $assert->pageTextContains('Sign in to continue');
 
     // Check that the redirect to limited access page has not been cached.
-    $this->drupalGet('<front>');
-    $assert->statusCodeEquals(200);
-    $this->drupalGet('/collections');
-    $assert->statusCodeEquals(200);
-    $this->drupalGet('/solutions');
-    $assert->statusCodeEquals(200);
-    $this->drupalGet('/keep-up-to-date');
-    $assert->statusCodeEquals(200);
-    $this->drupalGet('/search');
-    $assert->statusCodeEquals(200);
+    $this->assertAccess('<front>');
+    $this->assertAccess('/collections');
+    $this->assertAccess('/solutions');
+    $this->assertAccess('/keep-up-to-date');
+    $this->assertAccess('/search');
+
+    // Grant the user with 'unlimited access' permission.
+    $this->bypassReadOnlyConfig();
+    $rid = $this->createRole(['unlimited access']);
+    $this->restoreReadOnlyConfig();
+    $this->account->addRole($rid);
+    $this->account->save();
+
+    $this->drupalLogin($this->account);
+
+    // Try to navigate to pages that are not accessible to regular users.
+    $this->assertAccess('<front>');
+    $this->assertAccess('/collections');
+    $this->assertAccess('/solutions');
+    $this->assertAccess('/keep-up-to-date');
+    $this->assertAccess('/search');
```

Comment:
It is not clear why this is being tested, the documentation says that these pages are not accessible to regular users, but they are, and this is already tested above. Was it the idea to have pages here such as the moderator only pages? But the permission doesn't grant access to anything.

Reply:
This was clarified in a call with @claudiu-cristea, we will update the documentation with additional information.